implementation.js

```diff
@@ -1,9 +1,64 @@
 'use strict';
 var ES = require('es-abstract/es6');
 var supportsDescriptors = require('define-properties').supportsDescriptors;
+var has = require('has');
+var global = require('global-object');
+
+var parseIterable = function (arrayLike) {
+	var done = false;
+	var iterableResponse;
+	var tempArray = [];
+
+	if (arrayLike && typeof arrayLike.next === 'function') {
+		while (!done) {
+			iterableResponse = arrayLike.next();
+			if (
+				has(iterableResponse, 'value')
+				&& has(iterableResponse, 'done')
+			) {
+				if (iterableResponse.done === true) {
+					done = true;
+					break; // eslint-disable-line no-restricted-syntax
+
+				} else if (iterableResponse.done !== false) {
+					break; // eslint-disable-line no-restricted-syntax
+				}
+
+				tempArray.push(iterableResponse.value);
+			} else if (iterableResponse.done === true) {
+				done = true;
+				break; // eslint-disable-line no-restricted-syntax
+			} else {
+				break; // eslint-disable-line no-restricted-syntax
+			}
+		}
+	}
+
+	return done ? tempArray : false;
+};
+
+var hasSymbols = typeof Symbol === 'function' && typeof Symbol.iterator === 'symbol';
+var isNative = require('lodash.isnative');
+var isCallable = require('is-callable');
+
+var usingIterator = function (items) {
+    if (hasSymbols && isCallable(items[Symbol.iterator])) {
+        return items[Symbol.iterator]();
+    } else if (isCallable(items['@@iterator'])) {
+        return items['@@iterator']();
+    } else if ('Set' in global && 'Map' in global && isNative(Set) && isNative(Map) && isCallable(items.entries) && isCallable(items.values)) {
```

Comment:
Right - I'm saying that you cant use the `constructor` property, ever, for reliable anything. Not that I only want things to work with the native Map/Set.

In addition, have you tried this in browsers? There exist browsers where `Symbol.iterator` doesn't exist, and Set/Map lack `.forEach`, but `for..of` works - but `for..of` is a syntax error in older browsers.

This task is not a simple one, which is why it's still outstanding.